src/mds/mdstypes.h

```diff
@@ -773,10 +1137,11 @@ struct dentry_key_t {
     } else {
       // name_%x
       long long unsigned x = 0;
-      sscanf(key.c_str() + i + 1, "%llx", &x);
+      std::string x_str(key.substr(i+1));
+      sscanf(x_str.c_str(), "%llx", &x);
       sn = x;
     }  
-    nm = string(key.c_str(), i);
+    nm = key.substr(i);
```

Comment:
should be substr(0, i). otherwise, nm is "_head"

Reply:
thanks for catching that, fixed!